From 4ee1193154e159414f564f023d98a7965bd877d3 Mon Sep 17 00:00:00 2001
From: Cai YiWei <cyw@rock-chips.com>
Date: Wed, 30 Dec 2020 15:14:16 +0800
Subject: [PATCH] media: rockchip: ispp: fix panic for vmap at interrupt

kernel BUG at mm/vmalloc.c:1391!
[<b01f1f30>] (__get_vm_area_node) from [<b01f20c4>] (get_vm_area_caller+0x48/0x50)
[<b01f20c4>] (get_vm_area_caller) from [<b01f2394>] (vmap+0x34/0x64)
[<b01f2394>] (vmap) from [<b04b11e8>] (dma_buf_vmap+0x50/0xec)
[<b04b11e8>] (dma_buf_vmap) from [<b05afb2c>] (vb2_dc_vaddr+0x28/0x30)
[<b05afb2c>] (vb2_dc_vaddr) from [<b05f2014>] (rkispp_frame_end+0x2bc/0x3c8)
[<b05f2014>] (rkispp_frame_end) from [<b05f28a8>] (rkispp_isr+0x19c/0x434)
[<b05f28a8>] (rkispp_isr) from [<b05f589c>] (irq_hdl+0x58/0x60)

Change-Id: Ib4e3814ca77174cd31788f173cadaef01aaa6a4f
Signed-off-by: Cai YiWei <cyw@rock-chips.com>
---
 drivers/media/platform/rockchip/ispp/stream.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/platform/rockchip/ispp/stream.c b/drivers/media/platform/rockchip/ispp/stream.c
index f67a7b18fc67..b313104bf45b 100644
--- a/drivers/media/platform/rockchip/ispp/stream.c
+++ b/drivers/media/platform/rockchip/ispp/stream.c
@@ -1371,6 +1371,7 @@ static void rkispp_buf_queue(struct vb2_buffer *vb)
 
 	memset(isppbuf->buff_addr, 0, sizeof(isppbuf->buff_addr));
 	for (i = 0; i < cap_fmt->mplanes; i++) {
+		vb2_plane_vaddr(vb, i);
 		if (stream->isppdev->hw_dev->is_mmu) {
 			sgt = vb2_dma_sg_plane_desc(vb, i);
 			isppbuf->buff_addr[i] = sg_dma_address(sgt->sgl);
-- 
2.35.3

